ocaml/lambda/translcore.ml

```diff
@@ -390,7 +395,7 @@ and transl_exp0 ~in_new_scope ~scopes e =
         | Record_unboxed _ -> assert false
         | Record_float -> Psetfloatfield (lbl.lbl_pos, Assignment)
         | Record_extension _ ->
-          Psetfield (lbl.lbl_pos + 1, maybe_pointer newval, Assignment)
+          Psetfield (lbl.lbl_pos, maybe_pointer newval, Assignment)
```

Comment:
While working on the merge with local allocs, this line came up as a conflict, and I think it is in fact a bug. The following code segfaults the toplevel with this change:
```ocaml
type t = ..;;
type t += Foo of { mutable x : string };;
match Foo { x = "a" } with Foo z as r -> (z.x <- "b"; r) | _ -> assert false;;
```

Reply:
Indeed, it looks like it's a mistake that crept up when removing the tag and size info from the original PR.
I'll submit a fix.